lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -152,6 +152,10 @@ public function getAllMetadata()
      */
     public function getMetadataFor($className)
     {
+        if (strpos($className, 'class@anonymous') === 0) {
```

Comment:
should be `strpos($className, '@anonymous') !== false` to be forward compatible with PHP 8

Reply:
Not know how PHP 8.* represent anonymous class.
Added check only for '@anonymous'.